enumerable.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield(to_a[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    length.times do |x|
+      yield(self[x], x)
+    end
+  end
```

Comment:
### my_each_with_index

- [ ] when a block is given when self is an array returns the array itself after calling the given block once for each element in self
`array.my_each_with_index(&block)`
#### expected
`[3, 0, 0, 5, 1, 7, 8, 7, 4, 7, 6, 1, 0, 1, 1, 4, 6, 2, 7, 6, 1, 8, 8, 0, 3, 8, 6, 4, 6, 6, 0, 8, 1, 8..., 1, 7, 2, 7, 8, 0, 1, 8, 2, 0, 6, 3, 7, 3, 8, 5, 6, 0, 2, 4, 5, 2, 4, 1, 8, 8, 6, 6, 6, 7, 3, 8, 6]`
#### got
`100`

#### hint
- make sure you return the array at the end

- [ ] when self is range raises the following error `undefined local variable or method 'length'`

#### hint
- As length is not a method of range, look for other methods of finding the length of a range or casting range into an array.